### PR TITLE
test(fixtures): add npm compatibility fixtures for dist-tags, prerelease, engines/os/cpu

### LINKED ISSUE #137: fixtures: add representative npm compatibility fixture set
## Context

M5 needs offline fixtures for compatibility scenarios before implementation relies on real npm metadata.

## Scope

Add representative offline registry/install fixtures for optional dependencies, peer dependencies, engines, OS/CPU metadata, aliases, scoped packages, dist-tags, and prerelease edge cases.

## Done criteria

- Each fixture represents one scenario.
- Expected output is checked in.
- No fixture uses live npm as the test oracle.
- Tests copy install projects to temporary directories before mutation.

## Context to preserve

- Compatibility fixtures should follow the M5 contract classifications rather than creating new behavior by example.
- Use representative offline npm metadata; do not use live npm as the test oracle.
- Keep each fixture focused on one compatibility scenario.

## Test context

- Expected outputs should include resolved records, lockfile snapshots, diagnostics, or filesystem trees as appropriate.
- Mutating tests must copy fixtures to temporary directories before running.

## Dependencies

This issue is blocked by:
- #133 compat: define optionalDependencies contract before implementation
- #134 compat: preserve peerDependencies metadata without ordinary enqueue
- #135 compat: define peer dependency diagnostics separately from resolution
- #136 compat: define npm alias and scoped package edge cases

### PR DESCRIPTION
**Pull request by nerdchanii**
## Contract

Closes #137.

No SPEC change. This PR adds offline registry fixtures for the three M5 compatibility scenarios listed in #137 that were **not** already covered by existing fixtures:

| Scenario | Already covered? | This PR |
| --- | --- | --- |
| optionalDependencies (#133) | ✅ `registry/optional-preserve/` | — |
| peerDependencies (#134/#135) | ✅ `registry/peer-preserve/` | — |
| npm alias (#136) | ✅ resolver reject tests | — |
| scoped packages (#136/#171) | ✅ `@rpm-fixture/*` everywhere | — |
| **dist-tags edge cases** | ❌ | ✅ `registry/dist-tags-edge/` |
| **prerelease edge cases** | ❌ | ✅ `registry/prerelease-edge/` |
| **engines / os / cpu** | ❌ | ✅ `registry/engines-os-cpu-preserve/` |

Owning SPECs (unchanged, asserted by tests):
- `docs/specs/core/registry/SPEC.md` — dist-tag resolution order, dangling-tag rejection (#114), engines/os/cpu no-filter policy
- `docs/specs/core/semver/SPEC.md` — prerelease exclusion (no RPM-specific dialect)

## Changes

Three new `registry/` fixtures, each following the established `metadata/` + `expected/resolved-packages.txt` offline-registry contract:

1. **`dist-tags-edge/`** — `@rpm-fixture/tagged` exposes `latest`/`stable`/`legacy` tags; `@rpm-fixture/tag-consumer` requests it via the `legacy` dist-tag (resolves to `1.0.0`, not the `latest` target `1.2.0`). `@rpm-fixture/dangling-tag` has `latest` pointing at a version absent from `versions`, exercising the #114 rejection guard.

2. **`prerelease-edge/`** — `@rpm-fixture/prerelease` publishes `1.0.0` (stable) and `1.1.0-beta.1`. A consumer requesting `^1.0.0` resolves to the stable `1.0.0`, proving node-semver prerelease exclusion.

3. **`engines-os-cpu-preserve/`** — `@rpm-fixture/platform-strict` declares `engines.node >=99.0.0`, `os: ["plan9"]`, `cpu: ["ia64"]`. A consumer still resolves it normally, proving the registry boundary applies **no** platform filtering.

Four resolver tests consume the fixtures, mirroring the peer/optional preserve pattern (`FixtureMetadataProvider` + expected-list comparison).

## Validation

```
just format-check   # ok
just check          # ok (cargo check --locked --all-targets)
just lint           # ok (clippy strict, -D warnings)
just test           # ok — 185 passed (184 lib + 1 integration), 0 failed
audit-fixtures      # ok (fixture audit: ok)
fixture-smoke       # ok
```

The 4 new tests pass:
- `dist_tag_spec_resolves_to_tagged_version`
- `dangling_dist_tag_target_is_rejected_as_unsatisfied`
- `prerelease_version_is_excluded_from_normal_range_match`
- `engines_os_cpu_metadata_does_not_filter_version_selection`

`just validate` reports `agent_assets.claude_security=fail`, but that failure is caused by untracked, gitignored `.claude/settings.local.json` local allow-rules — it is pre-existing and unrelated to this change.

## Checklist

- [x] Each fixture represents one scenario.
- [x] Expected output is checked in.
- [x] No fixture uses live npm as the test oracle (all tarballs at `registry.example.invalid`).
- [x] Tests copy inputs through the existing `FixtureMetadataProvider` / temp-dir harness; original fixtures are never mutated.
- [x] No SPEC change required (existing contracts are asserted, not redefined).

### CODE REVIEW

File: src/core/resolver/mod.rs [hunk 1]
```diff
@@ -730,4 +730,179 @@ mod tests {
...
+            vec![DependencyRequest::new(
+                "@rpm-fixture/dangling-tag",
+                "latest",
+                DependencyRequestKind::DirectProduction,
+            )],
+            &provider,
+        )
+        .expect_err("a dist-tag pointing at an absent version must fail resolution");
+
+        assert!(matches!(error, ResolutionError::VersionSelection { .. }));
```
[the remark this answers is not in the document]
Comment (nerdchanii) [reply src/core/resolver/mod.rs]: Resolved in 4db492a. The regression assertion now matches ResolutionError::VersionSelection with nested SemverError::UnsatisfiedRange and requires range == "latest".

File: tests/fixtures/registry/engines-os-cpu-preserve/metadata/@rpm-fixture__platform-strict.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      "name": "@rpm-fixture/platform-strict",
+      "version": "1.0.0",
+      "description": "Fixture package declaring restrictive engines, os, and cpu metadata that must not trigger filtering at the registry boundary",
+      "dependencies": {},
+      "engines": {
+        "node": ">=99.0.0"
+      },
+      "os": [
+        "!win32",
+        "!darwin"
```
[the remark this answers is not in the document]
Comment (nerdchanii) [reply tests/fixtures/registry/engines-os-cpu-preserve/metadata/@rpm-fixture__platform-strict.json]: Resolved in 4db492a. The fixture now uses deliberately incompatible os ["plan9"] and cpu ["ia64"] values, so the no-filter coverage is independent of a Linux/x64 host.

**Review by nerdchanii — COMMENTED** [review]
### Code review

No issues found. Checked for correctness, contract integrity, filesystem
safety, and determinism.

🤖 Generated with [Claude Code](https://claude.ai/code)
